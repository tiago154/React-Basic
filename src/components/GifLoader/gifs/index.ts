export { default as pikachu } from './pikachu-running.gif'
export { default as pokeball } from './pokeball-catching.gif'
export { default as standing } from './pikachu-standing.gif'
