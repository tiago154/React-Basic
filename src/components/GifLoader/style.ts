import styled from 'styled-components'

interface IPropsStyled {
  width: number
}

export const Container = styled.div`
  height: auto;
`

export const Gif = styled.img`
    ${({ width }: IPropsStyled) => width && `width: ${width}px`};
    height: auto;
`
