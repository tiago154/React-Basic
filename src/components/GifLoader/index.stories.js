import React from 'react'
import GifLoader, { GifEnum } from '.'
import { withKnobs, select, number } from '@storybook/addon-knobs'

export default {
  title: 'Loading',
  decorators: [withKnobs]
}

export const Imagens = () => <GifLoader
  loadingType={select('loadingType', GifEnum, GifEnum.pikachu)}
  width={number('width', 250)}
/>
