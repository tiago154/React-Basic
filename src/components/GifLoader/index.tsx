import React from 'react'
import { Gif, Container } from './style'
import * as gifs from './gifs'

export enum GifEnum {
  'pikachu' = 'pikachu',
  'pokeball' = 'pokeball',
  'standing' = 'standing'
}
interface IProps {
  width: number,
  loadingType?: GifEnum
}

const GifLoader = ({ width, loadingType = GifEnum.pokeball }: IProps) =>
  (
    <Container>
      <Gif src={gifs[loadingType]} width={width} />
    </Container>
  )

export default GifLoader
