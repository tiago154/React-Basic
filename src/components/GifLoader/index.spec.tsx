import React from 'react'
import GifLoader, { GifEnum } from '.'
import * as gifs from './gifs'
import renderer from 'react-test-renderer'

describe('PokemonType', () => {
  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <GifLoader
          width={250}
          loadingType={GifEnum.standing}
        />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
