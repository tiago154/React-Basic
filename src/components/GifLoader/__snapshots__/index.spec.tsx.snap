// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonType Renders correctly 1`] = `
<div
  className="sc-bdVaJa dFcHew"
>
  <img
    className="sc-bwzfXH hvKFeJ"
    src="default"
    width={250}
  />
</div>
`;
